File: app/src/main/java/com/example/android_fragments_basics/MainActivity.java
package com.example.android_fragments_basics;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import java.util.ArrayList;
import java.util.Random;

public class MainActivity extends AppCompatActivity implements RandomItemFragment.OnListFragmentInteractionListener {
 RandomItem randomItem;
    public static ArrayList<RandomItem> randomList;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        randomList = new ArrayList<>(100);
        for(int i = 0; i<99;i++){
            randomItem = new RandomItem(String.valueOf(i));
            randomItem.setRandomInt(i);
            randomList.add(randomItem);

        }

    }
    @Override
    public void onListFragmentInteraction(RandomItem item) {
        DetailsFragment detailsFragment = new DetailsFragment();
    }
}

File: app/src/main/java/com/lambdaschool/congressfragmentsproject/activity/MainActivity.kt
package com.lambdaschool.congressfragmentsproject.activity

import android.graphics.Bitmap
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import com.lambdaschool.congressfragmentsproject.R
import com.lambdaschool.congressfragmentsproject.data.CongressDao
import com.lambdaschool.congressfragmentsproject.fragment.CongresspersonOverviewFragment
import com.lambdaschool.congressfragmentsproject.fragment.DetailsFragment
import com.lambdaschool.congressfragmentsproject.model.CongresspersonDetails
import com.lambdaschool.congressfragmentsproject.model.CongresspersonOverview
import kotlinx.android.synthetic.main.activity_main.*
import java.util.ArrayList

class MainActivity : AppCompatActivity(), CongresspersonOverviewFragment.OnListFragmentInteractionListener {

    companion object {
        const val FRAGMENT_KEY = "098HF18672G4FPIJH"
    }

    override fun onListFragmentInteraction(item: CongresspersonOverview?) {

        val fragment = DetailsFragment()

        val bundle = Bundle()
        bundle.putSerializable(FRAGMENT_KEY, item?.id)
        fragment.arguments = bundle

        supportFragmentManager.beginTransaction()
            .replace(R.id.main_activity_fragment_hodler, fragment)
            .addToBackStack(null)
            .commit()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // get an overview list for all members of congress
        val allMembers: ArrayList<CongresspersonOverview> = CongressDao.allMembers

        // get details for a single member of congress
        val singleMemberDetails: CongresspersonDetails? = allMembers[0].id?.let { CongressDao.getMemberDetails(it) }

        // get congressperson portrait
        val image: Bitmap? = allMembers[0].id?.let { CongressDao.getImage(it) }
    }
}